django_app/redbox_app/redbox_core/views/signup_views.py

```diff
@@ -0,0 +1,172 @@
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        form = SignUpForm(request.POST)
+
+        # Only allow .gov.uk email accounts
+        email = request.POST.get("email")
+        if not email.endswith(".gov.uk"):
+            form.add_error("email", "The email must be a valid gov.uk email account")
+
+        if form.is_valid():
+            request.session["sign_up_data"] = form.cleaned_data
+            return redirect("sign-up-page-2")
+        else:
+            return render(request, "sign-up-page-1.html", {"form": form})
+
+
+class Signup2(View):
+    def get(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        form = SignUpForm()
+        return render(request, "sign-up-page-2.html", {"form": form})
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        combined_data = {**request.session.get("sign_up_data", {}), **request.POST.dict()}
+        query_dict = QueryDict("", mutable=True)
+        query_dict.update(combined_data)
+        form = SignUpForm(query_dict)
+
+        if form.is_valid():
+            request.session["sign_up_data"] = form.cleaned_data
+            return redirect("sign-up-page-3")
+        else:
+            return render(request, "sign-up-page-2.html", {"form": form})
+
+
+class Signup3(View):
+    def get(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        form = SignUpForm()
+        return render(request, "sign-up-page-3.html", {"form": form})
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        combined_data = {**request.session.get("sign_up_data", {}), **request.POST.dict()}
+        query_dict = QueryDict("", mutable=True)
+        query_dict.update(combined_data)
+        form = SignUpForm(query_dict)
+
+        if form.is_valid():
+            request.session["sign_up_data"] = form.cleaned_data
+            return redirect("sign-up-page-4")
+        else:
+            return render(request, "sign-up-page-3.html", {"form": form})
+
+
+class Signup4(View):
+    def get(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        form = SignUpForm()
+        return render(request, "sign-up-page-4.html", {"form": form})
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        if os.getenv("ALLOW_SIGN_UPS") != "True":
+            return redirect("homepage")
+        combined_data = {**request.session.get("sign_up_data", {}), **request.POST.dict()}
+        query_dict = QueryDict("", mutable=True)
```

Comment:
```suggestion
        query_dict = QueryDict(combined_data)
```

i *think* this will work

Reply:
I've just tried it and it doesn't. That would have been nice. Any other suggestions @gecBurton?

Reply:
then just ignore my suggestion @KevinEtchells :)